services/email/app/main.py

```
@@ -78,77 +78,21 @@ def send_report():
-                 module_name))
-
-    status_summary = ' | '.join([f'{status_name}: {count}' for status_name, count in statuses.items()])
-    result_summary = ' | '.join([f'{result or "Not run"}: {count}' for result, count in results.items()])
-
-    creation = datetime.strptime(run['creation'], '%Y-%m-%dT%H:%M:%SZ') - timedelta(hours=8)
-
-    summaries = [('Id', run['id']),
-                 ('Creation', str(creation) + ' PST'),
-                 ('Creator', run['details'].get('a01.reserved.creator', 'N/A')),
-                 ('Remark', run['settings'].get('a01.reserved.remark', 'N/A')),
-                 ('Live', run['settings'].get('a01.reserved.livemode')),
-                 ('Task', status_summary),
-                 ('Image', run['settings'].get('a01.reserved.imagename', 'N/A')),
-                 ('Result', result_summary)]
-
-    content = f"""\
-    <html>
-        <body>
-            <div>
-                <h2>Summary</h2>
-                {tabulate(summaries, tablefmt="html")}
-            </div>
-            <div>
-                <h2>Failures</h2>
-                {tabulate(failure, headers=("id", "name", "status", "result", "duration(ms)", "module"), tablefmt="html")}
-            </div>
-            <div>
-                <h2>More details</h2>
-                <p>Install the latest release of A01 client to download log and recordings.</p>
-                <p>Instruction is here: https://github.com/azure/adx-automation-client</p>
-                <code>
-                $ a01 login<br>
-                $ a01 get runs -l {run['id']}<br>
-                </code>
-                <p>Contact: trdai@microsoft.com</p>
-            </div>
-        </body>
-    </html>"""
+    product = run['details'].get('a01.reserved.product', 'azurecli')
+    directory = f'{os.path.dirname(os.path.abspath(__file__))}/templates'
+
+    if not os.path.exists(f'{directory}/{product}.html'):
+        product = 'azurecli'
```

Make it fail or exit instead of falling back.

---

Fixed